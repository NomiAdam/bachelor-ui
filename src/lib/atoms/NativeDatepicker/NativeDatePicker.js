import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import DEFAULT_THEME from '../../utils/resolveTheme';

const StyledLabel = styled.label`
    display: block;
`;

const StyledInput = styled.input`
  ${ DEFAULT_THEME }
  padding: 3px 5px;
  box-shadow: inset 0 3px 6px rgba(0,0,0,0.1);
  width: 190px;
  &::-webkit-inner-spin-button {
    display: none;
  }
  &::-webkit-calendar-picker-indicator {
    opacity: 0;
  }
  background:#fff url(https://cdn1.iconfinder.com/data/icons/cc_mono_icon_set/blacks/16x16/calendar_2.png)  97% 50% no-repeat;
`;

const NativeDatePicker = ({
    onChange, value, label, min, max,
}) => (
    <div>
        <StyledLabel>{label}</StyledLabel>
        <StyledInput type="date" onChange={ onChange } value={ value } min={ min } max={ max } />
    </div>
);

NativeDatePicker.propTypes = {
    /**
     * Function to handle onChange event on Date input
     */
    onChange: PropTypes.func,
    /**
     * Current date value, format YYYY-MM-DD is expected
     */
    value: PropTypes.string,
    /**
     * String value of label above NativeDatePicker
     */
    label: PropTypes.string,
    /**
     * String value representing minimal allowed value of date, format YYYY-MM-DD is expected
     */
    min: PropTypes.string,
    /**
     * String value representing maximal allowed value of date, format YYYY-MM-DD is expected
     */
    max: PropTypes.string,
};

export { StyledInput };
export default NativeDatePicker;
