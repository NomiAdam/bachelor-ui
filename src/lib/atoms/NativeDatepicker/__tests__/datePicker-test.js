/* eslint-disable no-undef */
import React from 'react';
import NativeDatePicker from '../NativeDatePicker';
import { shallow } from 'enzyme';

const mock = jest.fn();

describe('NativeDatePicker', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });
    it('should be defined', () => {
        expect(shallow(<NativeDatePicker />)).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<NativeDatePicker />)).toMatchSnapshot();
    });
    it('should match the snapshot when labeled', () => {
        expect(shallow(<NativeDatePicker label={'Label'} />)).toMatchSnapshot();
    });
    it('should not fire action without click', () => {
        shallow(<NativeDatePicker onChange={mock}/>);
        expect(mock).not.toHaveBeenCalled();
    });
});
