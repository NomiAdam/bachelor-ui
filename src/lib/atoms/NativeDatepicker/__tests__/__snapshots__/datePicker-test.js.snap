// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NativeDatePicker should match the snapshot 1`] = `
<NativeDatePicker__StyledWrapper>
  <NativeDatePicker__StyledLabel />
  <NativeDatePicker__StyledInput
    type="date"
  />
</NativeDatePicker__StyledWrapper>
`;

exports[`NativeDatePicker should match the snapshot when labeled 1`] = `
<NativeDatePicker__StyledWrapper>
  <NativeDatePicker__StyledLabel>
    Label
  </NativeDatePicker__StyledLabel>
  <NativeDatePicker__StyledInput
    type="date"
  />
</NativeDatePicker__StyledWrapper>
`;
